engineio/src/client/socket.rs

```diff
@@ -290,24 +290,21 @@ impl Socket {
         self.socket.connect()
     }
 
+    /// Disconnects the connection.
+    pub fn disconnect(&self) -> Result<()> {
+        self.socket.disconnect()
+    }
+
     /// Sends a packet to the server.
     pub fn emit(&self, packet: Packet) -> Result<()> {
         self.socket.emit(packet)
     }
 
     /// Polls for next payload
-    pub(crate) fn poll(&self) -> Result<Option<Payload>> {
-        let payload = self.socket.poll()?;
-
-        if payload.is_none() {
-            return Ok(None);
-        }
-
-        let payload = payload.unwrap();
-
-        let iter = payload.iter();
-
-        for packet in iter {
+    //TODO: private?
```

Comment:
What do you think about `#[doc(hidden)]`? As far as I understood the poll method is needed by the socketio part, or am I wrong?

Reply:
Generally speaking: I think the iterator should be the go-to way of receiving packets as its nice and clean.

Reply:
> What do you think about `#[doc(hidden)]`? As far as I understood the poll method is needed by the socketio part, or am I wrong?

done

Reply:
> Generally speaking: I think the iterator should be the go-to way of receiving packets as its nice and clean.

Made socketio iter pub